### PR TITLE
Converts npc saves to jsons

### PR DESCRIPTION
**Pull request by Jordie0608**
All npc_saves files are now stored and loaded as jsons instead of .savs. 

No functional changes; I had to rewrite bits of the code for saving/loading secret satchels but it should work the same unless I misunderstood how they're meant to work.

Chisel messages and secret satchel files are now separated by map type again, this is to avoid loading excess data from other maps since the jsons are deleted before being written to.

Not extensively tested but each systems seemed to work when I tried it out.

@nfreader

### CODE REVIEW

**Comment by optimumtact** [comment]
all we neeed to fix up now is player saves and we're away !

**Comment by nfreader** [comment]
Exciting!

**Comment by optimumtact** [comment]
Does this convert the existing files once?

**Comment by Jordie0608** [comment]
No, I looked through the save data we have and didn't think it'd be worth saving if it meant having to keep around a proc to convert each file for downstreams.

**Comment by optimumtact** [comment]
it's so trivial though, you just retain the old loading behaviour

I don't understand this approach

**Comment by Jordie0608** [comment]
I'd rather be gone with it all, data and procs. The data isn't valuable and would be back to normal in a couple of days, and the old behaviour will become procs that can't be done away with.

**Comment by RemieRichards** [comment]
or just create a one time tool in dm..., open the .savs, dump out json, simple.

**Comment by optimumtact** [comment]
NO CONVERTORS NO MERGE

**Comment by Cyberboss** [comment]
woop woop

**Comment by optimumtact** [comment]
he just needs some australian time

**Comment by Jordie0608** [comment]
\>2 days
cmon

Fiiiiine you nerds I'll make the old ones convert.

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -21,67 +19,77 @@ SUBSYSTEM_DEF(persistence)
...
-	var/placed_satchels = 0
-
-	if(!isnull(old_secret_satchels))
-		expanded_old_satchels = splittext(old_secret_satchels,"#")
-		if(PlaceSecretSatchel(expanded_old_satchels))
-			placed_satchels++
+	var/placed_satchel = 0
+	var/path
+	var/obj/item/storage/backpack/satchel/flat/F = new()
+	if(fexists("data/npc_saves/SecretSatchels.sav"))
```
Comment (AnturK) [inline code/controllers/subsystem/persistence.dm]: Put in a comment that it'ss just for back compatibility, please.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**